pipecat-client-android-small-webrtc-transport/src/main/java/ai/pipecat/client/small_webrtc_transport/SmallWebRTCTransport.kt

```diff
@@ -0,0 +1,274 @@
+package ai.pipecat.client.small_webrtc_transport
+
+import ai.pipecat.client.result.Future
+import ai.pipecat.client.result.RTVIError
+import ai.pipecat.client.result.resolvedPromiseErr
+import ai.pipecat.client.result.resolvedPromiseOk
+import ai.pipecat.client.result.withPromise
+import ai.pipecat.client.transport.AuthBundle
+import ai.pipecat.client.transport.MsgClientToServer
+import ai.pipecat.client.transport.MsgServerToClient
+import ai.pipecat.client.transport.Transport
+import ai.pipecat.client.transport.TransportContext
+import ai.pipecat.client.transport.TransportFactory
+import ai.pipecat.client.types.MediaDeviceId
+import ai.pipecat.client.types.MediaDeviceInfo
+import ai.pipecat.client.types.Participant
+import ai.pipecat.client.types.ParticipantId
+import ai.pipecat.client.types.TransportState
+import android.annotation.SuppressLint
+import android.content.Context
+import android.media.AudioManager
+import android.util.Log
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+import kotlinx.serialization.json.decodeFromJsonElement
+
+
+private val BOT_PARTICIPANT = Participant(
+    id = ParticipantId("bot"),
+    name = null,
+    local = false
+)
+
+private val LOCAL_PARTICIPANT = Participant(
+    id = ParticipantId("local"),
+    name = null,
+    local = true
+)
+
+class SmallWebRTCTransport internal constructor(
+    private val transportContext: TransportContext,
+    androidContext: Context,
+    private val serverUrl: String,
+) : Transport() {
+
+    companion object {
+        private const val TAG = "SmallWebRTCTransport"
+    }
+
+    object AudioDevices {
+        val Speakerphone = MediaDeviceInfo(
+            id = MediaDeviceId("speakerphone"),
+            name = "Speakerphone"
+        )
+
+        val Earpiece = MediaDeviceInfo(
+            id = MediaDeviceId("earpiece"),
+            name = "Earpiece"
+        )
+    }
+
+    object Cameras {
+        val Front = MediaDeviceInfo(id = MediaDeviceId("cam-front"), name = "Front Camera")
+        val Rear = MediaDeviceInfo(id = MediaDeviceId("cam-rear"), name = "Rear Camera")
+    }
+
+    class Factory(
+        private val androidContext: Context,
+        private val serverUrl: String
+    ) : TransportFactory {
+        override fun createTransport(context: TransportContext): Transport {
+            return SmallWebRTCTransport(context, androidContext, serverUrl)
+        }
+    }
+
+    private var state = TransportState.Disconnected
+
+    private val appContext = androidContext.applicationContext
+    private val thread = transportContext.thread
+
+    private var client: WebRTCClient? = null
+    private var selectedCam = CameraMode.Front
+
+    override fun initDevices(): Future<Unit, RTVIError> = resolvedPromiseOk(thread, Unit)
```

Comment:
Are we going to implement it to initialize devices state and report initial available and selected devices ?
This is what I am [triggering on iOS](https://github.com/pipecat-ai/pipecat-client-ios-small-webrtc/blob/a6e4516b1fcbed772ca97a9616dddc9329097958/Sources/PipecatClientIOSSmallWebrtc/SmallWebRTCTransport.swift#L47).